akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -260,3 +248,24 @@ private[typed] class GuardianActorAdapter[T](_initialBehavior: Behavior[T]) exte
   case object Start
 
 }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[typed] final class ComposedStoppingBehavior[T](
+  lastBehavior: Behavior[T],
+  stopBehavior: StoppedBehavior[T]) extends ExtensibleBehavior[T] {
+  override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] =
+    throw new IllegalStateException("Stopping, should never receieve a message")
```

Comment:
> should never receieve a message

This is private[typed] vs user facing. We have to satisfy the parent requirements, disallow shooting oneself in the foot, but reacting with throwing versus returning what is declared, some kind of `Behavior[T]` ... can we perhaps return a `FailedBehavior` or a new `UnsupportedBehavior[T]` or `UnexpectedBehavior[T]` ?

Reply:
I agree with _fail fast_ but...perhaps there's another way.

Reply:
I think it would be a bug in Akka if it occured, and therefore the throwing

Reply:
Yeah, it would be a bug in Akka, not anything the user could cause.

Reply:
Sounds good